20-api-v2/00-intro.md

```diff
@@ -0,0 +1,252 @@
+StableSerialize(Foo) = 0a03coffee1202beaf
+SHA-512(StableSerialize(Foo)) =
+5efec2432616ca322824a7140d5ac332c6a3a388d2746f8cff6e48909d36829f9cb8586718d457c9540112d52ea2da2b448b2f6f689b3f5813c185b426267ed2
+
+R,S = Sign(private key, SHA-512(StableSerialize(Foo)), secp256r1)
+R = e13f3e71db728b85acc4cea688d3dae6b01453d2bff1b5ebc2695cedfef7fdd5
+S = 2ecbc0cc0ae4f70696682b4e358a4b698d74f9b708c13470e5c808fe04f526e5
+
+Signature =
+04e13f3e71db728b85acc4cea688d3dae6b01453d2bff1b5ebc2695cedfef7fdd52ecbc0cc0ae4f70696682b4e358a4b698d74f9b708c13470e5c808fe04
+```
+
+Signature function uses random number generator, which grants different
+signatures to the same message with each signing cycle.
+
+## Signature chaining in requests and responses
+
+The structure and the order of signatures for `Request` and `Response` messages
+are the same. Therefore, everything about `Request` message written below is
+true for `Response` on corresponding structures.
+
+All signatures are kept in `RequestVerificationHeader` structure which is filled
+before a message is sent. The \glspl{Node} involved in the transmission and resigning
+of the messages fill in their own structure copies and put them in origin field
+recursively.
+
+If a user sends a request directly to a receiver, the chain consists of one
+`RequestVerificationHeader` only.
+
+![Signature chain verification](pic/verification-chain)
+
+The `RequestVerificationHeader` always carries three signatures:
+
+* Message body signature
+* Meta header signature
+* Verification header signature
+
+### Message body signature
+
+Each RPC message has a field called `Body`. This structure is serialized
+stably; it is signed by the sender only. If a message is retransmitted, this
+signature is never set in the new copies of `VerificationHeader`.
+
+![Body signature verification](pic/verification-body-chain)
+
+### Meta header signature
+
+Each RPC meassage has a field called meta_header. Meta headers are changed for
+every retransmission (e.g. TTL is reduced) and form an equivalent chain.
+meta_signature field contains the signature for an already organized structure
+of the meta header.
+
+![Meta header signature verification](pic/verification-meta)
+
+### Verification header signature
+
+While putting a new verification header, intermediate nodes should sign the
+preceding verification header and put the signature in the origin_signature
+field. The requestor does not set this signature.
+
+![Verification header signature verification](pic/verification-origin)
+
+## Container service signatures
+
+In addition to the RPC requests themselves there is a need to sign followong structures:
+
+* Container in `container.PutRequest.Body` message
+* Container ID in `container.DeleteRequest.Body` message
+* Extended ACL table structure in `container.SetExtendedACLRequest.Body` message
+
+Those structures' signature is verified by Smart Contracts, hence it must be
```

Comment:
lower case for `smart contract`

Reply:
This PR is focused on intro part, as the rest of API documentation is auto-generated. It will be updated with following `neofs-api` patch releases.

Reply:
fixed